portal-bridge/src/bridge/history.rs

```diff
@@ -149,26 +149,27 @@ impl HistoryBridge {
         let latest_block = self.execution_api.get_latest_block_number().await.expect(
             "Error launching bridge in backfill mode. Unable to get latest block from provider.",
         );
-        let (looped, mode_type) = match self.mode.clone() {
+        let (is_single_mode, mode_type) = match self.mode.clone() {
             BridgeMode::Backfill(val) => (true, val),
             BridgeMode::Single(val) => (false, val),
             _ => panic!("Invalid backfill mode"),
         };
-        let (mut start_block, mut end_block, mut epoch_index) = match mode_type {
-            // end block will be same for an epoch in single & backfill modes
-            ModeType::Epoch(epoch_number) => (
-                epoch_number * EPOCH_SIZE,
-                ((epoch_number + 1) * EPOCH_SIZE),
-                epoch_number,
-            ),
+        let (start_block, mut end_block) = match mode_type {
+            ModeType::Epoch(epoch_number) => {
+                let end_block = match is_single_mode {
+                    true => (epoch_number + 1) * EPOCH_SIZE,
+                    false => latest_block,
+                };
+                (epoch_number * EPOCH_SIZE, end_block)
+            }
             ModeType::Block(block) => {
-                let epoch_index = block / EPOCH_SIZE;
-                let end_block = match looped {
-                    true => (epoch_index + 1) * EPOCH_SIZE,
-                    false => block + 1,
+                let end_block = match is_single_mode {
+                    true => block + 1,
+                    false => latest_block,
                 };
-                (block, end_block, epoch_index)
+                (block, end_block)
             }
+            ModeType::BlockRange(start_block, end_block) => (start_block, end_block + 1),
```

Comment:
Tbh, I think it would be a worthwhile improvement to extract this logic into a method on `BridgeMode`. Looks like the only external variable needed is `latest_block` which could be easily passed in. It would also be good for sanity's sake to add some unit testing to this logic. If you want to address it in a following pr, that's fine by me.

Reply:
I think it should go in a follow up PR yeah I made a issue for it to track it https://github.com/ethereum/trin/issues/1100